### PR TITLE
fix: remove cancel button if user cannot cancel job

### LINKED ISSUE #11449: disable `Cancel` button entirely for non-Owners
we have a template option to allow users to cancel in-progress workspace jobs. if this is disabled, the cancel button is still shown. when clicked, an error message appears saying only owners are allowed to cancel the build.

instead of returning an error, we should disable or remove the cancel button entirely for users who are non-owners (assuming the template option is disabled). this is feedback from one of our large customers.

Screenshot 2024-01-05 at 09 59 36

### PR DESCRIPTION
**Pull request by f0ssel**
Closes https://github.com/coder/coder/issues/11449

The user can only cancel under the following circumstances according to the backend:
1. You have the owner role.
OR
2. The template specifies that ANY user can cancel the builds based on it.  

Owner View if user cancelation is disabled:
Screenshot 2024-01-10 at 11 42 10 AM

Member View if user cancelation is disabled:
Screenshot 2024-01-10 at 11 41 50 AM

### CODE REVIEW

File: site/src/pages/WorkspacePage/Workspace.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ export interface WorkspaceProps {
   workspaceErrors: WorkspaceErrors;
   buildInfo?: TypesGen.BuildInfoResponse;
   sshPrefix?: string;
-  template?: TypesGen.Template;
+  template: TypesGen.Template;
```
Comment (Kira-Pilot) [inline site/src/pages/WorkspacePage/Workspace.tsx:58]: Why did we change this type?

**Review by Kira-Pilot — COMMENTED** [review]
Thanks for making this change! Looks good to me.

Can we add a storybook test before we merge it up? Your screenshots would be good test cases.
~You'll need to mock the user value the `useMe` hook returns and the way to do this is to mock out the `AuthContext.Provider`~

~[Here's an example](https://github.com/coder/coder/blob/main/site/src/pages/WorkspacesPage/WorkspacesPageView.stories.tsx#L144) of how we mock providers in a Storybook file.~

As discussed over Slack, we'll need to move this hook invocation up to fix the broken stories. Since you'll be accessing `isOwner` as a prop, you won't need to mock that auth provider.

File: site/src/pages/WorkspacePage/Workspace.tsx (hunk 1 above)
Comment (f0ssel) [reply site/src/pages/WorkspacePage/Workspace.tsx:58]: I noticed it didn't need to be conditional and see those as a liability so I just drive-by fixed it while I was reading the code.

**Review by BrunoQuaresma — APPROVED** [review]
Looks good!

**Review by Kira-Pilot — APPROVED** [review]

**Pull request merged**